src/world_interface/motor/can_motor.cpp

```diff
@@ -0,0 +1,36 @@
+#include "can_motor.h"
+
+#include "../../CAN/CAN.h"
+#include "../../CAN/CANMotor.h"
+#include "../../CAN/CANUtils.h"
+#include "../real_world_constants.h"
+
+namespace robot {
+can_motor::can_motor(robot::types::motorid_t motor, bool hasPosSensor)
+	: base_motor(motor, hasPosSensor) {}
+
+void can_motor::setMotorPower(double power) {
+	// unschedule velocity event if exists
+	unscheduleVelocityEvent();
+
+	can::deviceserial_t serial = motorSerialIDMap.at(motor_id);
+	auto& scaleMap = power < 0 ? negative_pwm_scales : positive_pwm_scales;
+	auto entry = scaleMap.find(motor_id);
+	if (entry != scaleMap.end()) {
+		power *= entry->second;
+	}
+	can::motor::setMotorPower(serial, power);
```

Comment:
Can we also move the logic of `ensureMotorMode` from `real_world_interface` to here? The motor mode should just be a field of this class, and we should do the same logic of setting the motor mode if required.

After that, we can get rid of the motor mode map in `real_world_interface`.

And remember to ensure motor modes for the setPos and setVel methods too. For the latter, override setVel, ensure the motor mode, and then invoke the base class implementation.